File: ros2_ws/src/ros2_app_cpp/src/number_counter.cpp
#include "rclcpp/rclcpp.hpp"

class NumberCounterNode : public rclcpp::Node 
{
public:
    NumberCounterNode() : Node("number_counter") 
    {
        RCLCPP_INFO(this->get_logger(), "Number counter has been started");
    }

private:
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto node = std::make_shared<NumberCounterNode>(); 
    rclcpp::spin(node);
    rclcpp::shutdown();
    return 0;
}

File: ros2_ws/src/ros2_app_py/ros2_app_py/number_counter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import Int64


class NumberCounterNode(Node): 
  def __init__(self):
    super().__init__("number_counter")
    self.counter_ = 0
    self.number_subscriber_ = self.create_subscription(
      Int64, "number", self.callback_number, 10)
    self.get_logger().info("Number counter has been started.")

  def callback_number(self, msg):
    self.counter_ += msg.data
    self.get_logger().info("Counter: "+ str(self.counter_))

def main(args=None):
  rclpy.init(args=args)
  node = NumberCounterNode() 
  rclpy.spin(node)
  rclpy.shutdown()


if __name__ == "__main__":
  main()
